### PR TITLE
completion: Don't show builtins on default rules

### PR DESCRIPTION
**Pull request by charlieegan3**
Related to https://github.com/StyraInc/regal/issues/1010

### CODE REVIEW

File: internal/lsp/completions/providers/builtins.go [hunk 1]
```diff
@@ -29,6 +29,11 @@ func (*BuiltIns) Run(c *cache.Cache, params types.CompletionParams, _ *Options)
 		return []types.CompletionItem{}, nil
 	}
 
+	// default rules cannot contain calls
+	if strings.HasPrefix(strings.TrimSpace(currentLine), "default") {
```
Comment (srenatus) [inline internal/lsp/completions/providers/builtins.go:33]:
```suggestion
	if strings.HasPrefix(strings.TrimSpace(currentLine), "default ") {
```
I could call my rule `defaultrule` and I'd be surprised why there are no builtin completions...? Or am I misreading  this? 😅

Comment (charlieegan3) [reply internal/lsp/completions/providers/builtins.go:33]: I think that makes sense yeah!

**Review by srenatus — APPROVED** [review]
Thanks

File: internal/lsp/completions/providers/builtins_test.go [hunk 2]
```diff
@@ -184,6 +184,39 @@ allow if c`
 	}
 }
 
+func TestBuiltIns_noDefeaultRule(t *testing.T) {
```
Comment (charlieegan3) [inline internal/lsp/completions/providers/builtins_test.go:187]:
```suggestion
func TestBuiltIns_noDefaultRule(t *testing.T) {
```

**Pull request merged**